### ISSUE TITLE
[Bug] Device Time Issue

### ISSUE THREAD

**Issue by StarWalkin** [google-research/android_world#129]
Hi, thanks for your impressive project. While running the benchmark I encountered with several issues concerning the device time.
- 'No Internet Connection' because device time is incorrect: It seems that the device time will be initialized before running every benchmark task, and it can cause 'No Internet Connection' in chrome, as the screenshot shows.
![image](https://github.com/user-attachments/assets/9108e2fa-82d4-430d-b9e8-ecf2e434e882)
Though most of the tasks don't require internet connection, the information may be a distraction to GUI agents(e.g. they may try really hard to fix the internet issue first).
- Task date mismatch. For some tasks like \'SportsTrackerActivityDuration\', a specific time will be assigned to instantiate the task template. However, I often find the generated task objective is inconsistent with the device time. For example, the task here is "How long was my skiing activity October 12 2023 in the OpenTracks app? Express your answer in minutes as a single integer." And in the app the only skiing activity is in yesterday, October 14th(the device time is Oct 15th). This makes the agent confused about where to find the required activity.
![image](https://github.com/user-attachments/assets/2f6bcc05-257f-43c2-bc51-68737e44d86f)

So how can I avoid these issues? Thanks!

**Comment by crawles** [comment]
Hi! Thanks for reporting. we will take a look. Can you please confirm that you’re using a Pixel 6 emulator?

**Comment by StarWalkin** [comment]
> Hi! Thanks for reporting. we will take a look. Can you please confirm that you’re using a Pixel 6 emulator?

Yes, I exactly followed the steps in README and the setup video :)

**Comment by StarWalkin** [comment]
If I turn on 'Set time automatically' in the system setting, it will be able to connect to the internet. But the time will be reset to October 2023 when running evaluation.
<img width="409" alt="image" src="https://github.com/user-attachments/assets/f6f04285-395a-4eec-a2b0-13341ee9d641">

**Comment by clink42** [comment]
For the sports tracker issue, I see a skiing activity on Thursday (5th one down). Is that not on October 12?

**Comment by crawles** [comment]
@StarWalkin , I have submitted a fix for the browser date time mismatch. Now for browser tasks, we set the date time as the current date time, which will remove that Chrome warning.
Regarding the Sports Tracker issue, as @clink42 mentions, in this case the answer should be the 5th row labeled "Outdoor Adv...", so this appears to be working as intended. Please let us know if you still believe there is a bug. Thank you.

**Comment by StarWalkin** [comment]
@crawles @clink42 I'm very sorry for mistakenly raising part of the issue due to my carelessness. The time-related problems have all been resolved now. Thank you very much for your prompt response and fix!

**Comment by crawles** [comment]
So glad to hear and no worries!
